static/app/components/notificationActions/notificationActionItem.tsx

```diff
@@ -0,0 +1,358 @@
+    };
+  }
+
+  function renderNotificationActionForm() {
+    if (serviceType === NotificationActionService.SENTRY_NOTIFICATION) {
+      // No form for Sentry notifications, just Cancel + Save buttons
+      return (
+        <NotificationActionFormContainer>
+          <NotificationActionCell>{renderDescription()}</NotificationActionCell>
+          <NotificationActionButtonContainer>
+            <Button onClick={handleCancel} size="xs">
+              {t('Cancel')}
+            </Button>
+            <Button priority="primary" size="xs" onClick={handleSave}>
+              {t('Save')}
+            </Button>
+          </NotificationActionButtonContainer>
+        </NotificationActionFormContainer>
+      );
+    }
+    if (serviceType === NotificationActionService.SLACK) {
+      return (
+        <SlackForm
+          action={updatedAction}
+          onChange={handleChange}
+          onSave={handleSave}
+          onCancel={handleCancel}
+          availableActions={availableActions}
+        />
+      );
+    }
+    if (serviceType === NotificationActionService.PAGERDUTY) {
+      return (
+        <PagerdutyForm
+          action={updatedAction}
+          onChange={handleChange}
+          onSave={handleSave}
+          onCancel={handleCancel}
+          pagerdutyIntegrations={pagerdutyIntegrations}
+        />
+      );
+    }
+    // TODO(enterprise): forms for email, msteams, sentry_app
+    return null;
+  }
+
+  return (
+    <StyledCard isEditing={isEditing}>
+      {isEditing ? (
+        <NotificationActionContainer>
+          <IconContainer>{renderIcon()}</IconContainer>
+          {renderNotificationActionForm()}
+        </NotificationActionContainer>
+      ) : (
+        <Fragment>
+          <NotificationActionContainer>
+            <IconContainer>{renderIcon()}</IconContainer>
+            <NotificationActionCell>{renderDescription()}</NotificationActionCell>
+          </NotificationActionContainer>
+          {renderEditButton()}
+        </Fragment>
+      )}
+    </StyledCard>
+  );
+}
+
+const StyledCard = styled(Card)<{isEditing: boolean}>`
+  padding: ${space(1)} ${space(1.5)};
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+  margin-bottom: ${space(1)};
+  background-color: ${props => (props.isEditing ? props.theme.surface200 : 'inherit')};
+`;
+
+const IconContainer = styled('div')`
+  margin-right: ${space(0.5)};
+  display: flex;
+  align-items: center;
+`;
+
+const NotificationActionContainer = styled('div')`
+  display: flex;
+  align-items: center;
+  width: 100%;
+`;
+
+export const NotificationActionCell = styled('div')`
+  display: flex;
+  align-items: center;
+  flex-wrap: wrap;
+  > * {
+    margin-left: ${space(0.5)};
+    margin-right: ${space(0.5)};
+  }
+`;
+
+export const NotificationActionFormContainer = styled('div')`
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+`;
+
+export const NotificationActionButtonContainer = styled('div')`
```

Comment:
This should probably be [ButtonBar](https://github.com/getsentry/sentry/blob/master/static/app/components/buttonBar.tsx)